stingray/io.py

```diff
@@ -92,108 +131,218 @@ def _get_additional_data(lctable, additional_columns):
+    t_start : float
+        Start time in Mission Epoch Time
+    t_stop : float
+        Stop time in Mission Epoch Time
+    pi_list : array-like
+        Raw Instrument energy channels
+    cal_pi_list : array-like
+        Calibrated PI channels (those that can be easily converted to energy
+        values, regardless of the instrument setup.)
+    energy_list : array-like
+        Energy of each photon in keV (only for NuSTAR, NICER, XMM)
+    instr : str
+        Name of the instrument (e.g. EPIC-pn or FPMA)
+    mission : str
+        Name of the instrument (e.g. XMM or NuSTAR)
+    mjdref : float
+        MJD reference time for the mission
+    header : str
+        Full header of the FITS file, for debugging purposes
+    detector_id : array-like, int
+        Detector id for each photon (e.g. each of the CCDs composing XMM's or
+        Chandra's instruments)
     """
+    from astropy.io import fits as pf
 
-    gtistring = assign_value_if_none(gtistring, 'GTI,STDGTI')
-    lchdulist = fits.open(fits_file)
+    gtistring = assign_value_if_none(gtistring, "GTI,GTI0,STDGTI")
+    lchdulist = pf.open(fits_file)
 
     # Load data table
     try:
         lctable = lchdulist[hduname].data
-    except:  # pragma: no cover
-        logging.warning('HDU %s not found. Trying first extension' % hduname)
+    except Exception:  # pragma: no cover
+        warnings.warn("HDU %s not found. Trying first extension" % hduname)
         lctable = lchdulist[1].data
+        hduname = 1
 
     # Read event list
     ev_list = np.array(lctable.field(column), dtype=np.longdouble)
-
+    detector_id = _get_detector_id(lctable)
+    det_number = None if detector_id is None else list(set(detector_id))
+    header = lchdulist[1].header
     # Read TIMEZERO keyword and apply it to events
     try:
-        timezero = np.longdouble(lchdulist[1].header['TIMEZERO'])
-    except:  # pragma: no cover
-        logging.warning("No TIMEZERO in file")
-        timezero = np.longdouble(0.)
+        timezero = np.longdouble(header["TIMEZERO"])
+    except Exception:  # pragma: no cover
+        warnings.warn("No TIMEZERO in file", AstropyUserWarning)
+        timezero = np.longdouble(0.0)
+
+    try:
+        instr = header["INSTRUME"].lower()
+    except Exception:
+        instr = "unknown"
+    try:
+        mission = header["TELESCOP"].strip().lower()
+    except Exception:
+        mission = "unknown"
 
     ev_list += timezero
 
     # Read TSTART, TSTOP from header
     try:
-        t_start = np.longdouble(lchdulist[1].header['TSTART'])
-        t_stop = np.longdouble(lchdulist[1].header['TSTOP'])
-    except:  # pragma: no cover
-        logging.warning("Tstart and Tstop error. using defaults")
+        t_start = np.longdouble(header["TSTART"])
+        t_stop = np.longdouble(header["TSTOP"])
+    except Exception:  # pragma: no cover
+        warnings.warn(
+            "Tstart and Tstop error. using defaults", AstropyUserWarning
```

Comment:
A well-behaved fits file should have all this information, at least for all the missions I use. I thought to add those warnings because they will be fired when the file is from some unknown mission (for which our machinery might produce strange results!) or simulations. Do you have any counter-example?

Reply:
Don't know. Maybe a fits file produced by a custom pipeline that doesn't return all the information?

Reply:
I eliminated TIMEZERO